### PR TITLE
Better ipfs

### PR DESCRIPTION
**Pull request by codinpsycho**
Better reads, index all NFTs from a wallet and using nftstorage.link on create
Other null checks and improvements

### CODE REVIEW

File: src/modules/db/remote-data-fetcher/data-fetcher.service.ts [hunk 1]
```diff
@@ -110,11 +108,12 @@ export class RemoteDataFetcherService {
...
   private async getOffChainMetadata(uri: string): Promise<any> {
-    const uriRes = await this.httpService.get(uri).toPromise();
-    if (uriRes.status != 200) {
-      throw new HttpException('Incorrect URI path', HttpStatus.INTERNAL_SERVER_ERROR);
+    try {
+      const uriRes = await this.httpService.get(uri).toPromise();
+      return uriRes.status === 200 ? uriRes.data : {};
+    } catch (error) {
+      //Dont throw exception, else all other gets will fail needlessly
+      console.log('error occurred for', uri);
```
Comment (vgvishesh) [inline src/modules/db/remote-data-fetcher/data-fetcher.service.ts:116]: We should throw exception here, and how to handle it or not should be decided by the caller function.

File: src/modules/nft/components/create-nft/dto/create-nft.dto.ts [hunk 2]
```diff
@@ -55,7 +55,7 @@ export class CreateNftDto {
 
   @ApiProperty({
     title: 'attributes',
-    type: Object,
+    type: String,
```
Comment (vgvishesh) [inline src/modules/nft/components/create-nft/dto/create-nft.dto.ts:58]: This is not a string, it is a json array.

File: src/modules/nft/components/read-nft/read-nft.service.ts [hunk 3]
```diff
@@ -15,14 +15,37 @@ export class ReadNftService {
...
+
+      const dbFilter = { owner: address, network: network };
+
+      if (update_authority) {
+        dbFilter['update_authority'] = update_authority;
+      }
+
+      const dbNftInfo = await this.nftInfoAccessor.find(dbFilter);
 
       //Right now its being fetched from chain only
```
Comment (vgvishesh) [inline src/modules/nft/components/read-nft/read-nft.service.ts:27]: Remove this comment.

**Review by vgvishesh — CHANGES_REQUESTED** [review]

File: src/modules/nft/components/create-nft/dto/create-nft.dto.ts (hunk 2 above)
Comment (codinpsycho) [reply src/modules/nft/components/create-nft/dto/create-nft.dto.ts:58]: Remember Abhirup had to stringify it.

**Review by vgvishesh — APPROVED** [review]

**Pull request merged**